shared/src/k8s/mod.rs

```diff
@@ -621,6 +621,69 @@ impl KubeInterface for KubeImpl {
     }
 }
 
+/// This deletes an Instance unless it has already been deleted by another node
+pub async fn try_delete_instance(
+    kube_interface: &impl KubeInterface,
+    instance_name: &str,
+    instance_namespace: &str,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    match kube_interface
```

Comment:
can there be 1 implementation for both delete functions?

Reply:
I am not sure how to do this since I cannot think off how to easily convert `&impl KubeInterface` into `Box<dyn KubeInterface>`. That makes it hard to have an `inner_try_delete_instance`.

Reply:
Figured out how to cast to make everything work. Fix coming soon

Reply:
Fixed in bbe2fb815